trailmet/algorithms/prune/network_slimming_prune.py

```diff
@@ -0,0 +1,232 @@
+                            print('In shape: {:d} Out shape:{:d}'.format(idx0.shape[0], idx1.shape[0]))
+                            w = m0.weight.data[:, idx0, :, :].clone()
+                            w = w[idx1, :, :, :].clone()
+                            m1.weight.data = w.clone()
+                        elif isinstance(m0, nn.Linear):
+                            idx0 = np.squeeze(np.argwhere(np.asarray(start_mask.cpu().numpy())))
+                            m1.weight.data = m0.weight.data[:, idx0].clone()
+
+                elif(args.arch=='resnet'):
+                    old_modules = list(model.modules())
+                    new_modules = list(newmodel.modules())
+                    layer_id_in_cfg = 0
+                    start_mask = torch.ones(3)
+                    end_mask = cfg_mask[layer_id_in_cfg]
+                    conv_count = 0
+
+                    for layer_id in range(len(old_modules)):
+                        m0 = old_modules[layer_id]
+                        m1 = new_modules[layer_id]
+                        if isinstance(m0, nn.BatchNorm2d):
+                            idx1 = np.squeeze(np.argwhere(np.asarray(end_mask.cpu().numpy())))
+                            if idx1.size == 1:
+                                idx1 = np.resize(idx1,(1,))
+
+                            if isinstance(old_modules[layer_id + 1], channel_selection):
+                                # If the next layer is the channel selection layer, then the current batchnorm 2d layer won't be pruned.
+                                m1.weight.data = m0.weight.data.clone()
+                                m1.bias.data = m0.bias.data.clone()
+                                m1.running_mean = m0.running_mean.clone()
+                                m1.running_var = m0.running_var.clone()
+
+                                # We need to set the channel selection layer.
+                                m2 = new_modules[layer_id + 1]
+                                m2.indexes.data.zero_()
+                                m2.indexes.data[idx1.tolist()] = 1.0
+
+                                layer_id_in_cfg += 1
+                                start_mask = end_mask.clone()
+                                if layer_id_in_cfg < len(cfg_mask):
+                                    end_mask = cfg_mask[layer_id_in_cfg]
+                            else:
+                                m1.weight.data = m0.weight.data[idx1.tolist()].clone()
+                                m1.bias.data = m0.bias.data[idx1.tolist()].clone()
+                                m1.running_mean = m0.running_mean[idx1.tolist()].clone()
+                                m1.running_var = m0.running_var[idx1.tolist()].clone()
+                                layer_id_in_cfg += 1
+                                start_mask = end_mask.clone()
+                                if layer_id_in_cfg < len(cfg_mask):  # do not change in Final FC
```

Comment:
there are a lot of parts of this code which can be made more structured. see chipnet.py and follow similar guidelines.

Reply:
Sir I have made some changes could you please check once.